frontend/src/components/pages/flatPage/FlatPage.tsx

```diff
@@ -60,27 +61,56 @@ export const FlatPageUI: React.FC<FlatPageUIProps> = ({ flatPageUrl }) => {
               </TextWithShadow>
             </div>
           )}
-          <div className="px-4 desktop:px-10vw py-4 desktop:py-10" id="flatPage_content">
+          <div className="px-4 mx-auto max-w-[900px]" id="flatPage_content">
             {(flatPage.attachment == null || flatPage.attachment.length === 0) && (
               <div className="flex justify-center py-6 desktop:py-12">
                 <h1 className="text-H3 desktop:text-H1 font-bold text-primary1 text-center">
                   {flatPage.title}
                 </h1>
               </div>
             )}
-            <div className="ml-5">
-              <Breadcrumb
-                breadcrumb={[
-                  {
-                    label: intl.formatMessage({ id: 'header.home' }),
-                    link: '/',
-                  },
-                  { label: flatPage?.title },
-                ]}
-              />
-            </div>
+            <Breadcrumb
+              breadcrumb={[
+                {
+                  label: intl.formatMessage({ id: 'header.home' }),
+                  link: '/',
+                },
+                { label: flatPage?.title },
+              ]}
+            />
             {flatPage.content !== null && flatPage.content.length > 0 && (
-              <HtmlText className="mb-10">{parse(flatPage.content)}</HtmlText>
+              <div className="custo-page-WYSIWYG mb-10 text-lg desktop:text-xl">
+                {parse(flatPage.content, {
+                  replace: (domNode: DOMNode) => {
+                    if (
+                      domNode instanceof Element &&
+                      domNode.attribs &&
+                      'data-ids' in domNode.attribs &&
+                      'data-type' in domNode.attribs &&
+                      domNode.attribs.class.includes('suggestions')
+                    ) {
+                      const suggestion = activitySuggestions.find(
+                        item =>
+                          item.results.map(({ id }) => id).join(',') ===
+                            domNode.attribs['data-ids'] &&
+                          item.type === getSuggestionType(domNode.attribs['data-type']),
+                      );
+                      if (!suggestion || suggestion.results.length === 0) {
+                        return null;
+                      }
+                      return (
+                        <HomeSection
+                          title={suggestion.titleTranslationId}
+                          iconUrl={suggestion.iconUrl}
+                          results={suggestion.results}
+                          asColumn
+                        />
+                      );
+                    }
+                    return domNode;
+                  },
+                })}
+              </div>
```

Comment:
Should this `parse()` function be called (and memoized) ahead of component render?

Something like:

```js
const parsedFlatPage = React.useMemo(
  () => {
    if (!flatPage.content || !flatPage.content.length) {
      return null;
    }

    return parse(flatPage.content, { replace: () => {/* … */} });
  },
  [flatPage.content],
);
```

then in component render:

```jsx
   {parsedFlatPage}
```


Also, replace function could be factorized too.

Reply:
Thx for your review :pray: